eng/Subsets.props

```diff
@@ -121,7 +121,8 @@
     <!-- CLR NativeAot only builds in a subset of the matrix -->
     <_NativeAotSupportedOS Condition="'$(TargetOS)' == 'windows' or '$(TargetOS)' == 'linux' or '$(TargetOS)' == 'osx' or '$(TargetOS)' == 'maccatalyst' or '$(TargetOS)' == 'iossimulator' or '$(TargetOS)' == 'ios' or '$(TargetOS)' == 'tvossimulator' or '$(TargetOS)' == 'tvos' or '$(TargetOS)' == 'freebsd'">true</_NativeAotSupportedOS>
     <_NativeAotSupportedArch Condition="'$(TargetArchitecture)' == 'x64' or '$(TargetArchitecture)' == 'arm64' or '$(TargetArchitecture)' == 'arm' or ('$(TargetOS)' == 'windows' and '$(TargetArchitecture)' == 'x86')">true</_NativeAotSupportedArch>
-    <NativeAotSupported Condition="'$(_NativeAotSupportedOS)' == 'true' and $(_NativeAotSupportedArch) == 'true'">true</NativeAotSupported>
+    <_NativeAotSupportedCrossOS Condition="'$(CrossBuild)' != 'true' or '$(TargetOS)' == '$(HostOS)' or '$(TargetsLinuxBionic)' == 'true' or '$(TargetsAppleMobile)' == 'true'">true</_NativeAotSupportedCrossOS>
+    <NativeAotSupported Condition="'$(_NativeAotSupportedOS)' == 'true' and '$(_NativeAotSupportedArch)' == 'true' and '$(_NativeAotSupportedCrossOS)' == 'true'">true</NativeAotSupported>
```

Comment:
Yup, we had `NativeAotSupported` definitions scattered across four places, this file and other three. I just consolidated them here.

The additional conditions from other places are: (after applying the OS/arch filters) NativeAotSupported is true for non- crossbuild or (crossbuild and hostos == targetos) or (crossbuild and islinuxbionic) or (crossbuild and appleplatforms).

Reply:
Wasn't the scattered definition just sharpening the NativeAotSupported from "we can build native AOT subset of the repo" to "we can build native AOT subset of the repo _and_ we can execute ILC as part of the repo build"? This was done so that e.g. we can build native AOT runtime on Linux, targeting FreeBSD, but we don't try to native AOT compile the crossgen.

Would we no longer build native AOT when doing a crossbuild from Linux to FreeBSD with this? (I picked FreeBSD, but I have not clue in what state it is, so it's just an example).

Reply:
I have made the intent clear by separating the concerns; `NativeAotCanExecuteOnHost` is its own property.